File: server/app/models/db/user/user_course_progress_requests/create_user_course_progress_request.py
from pydantic import BaseModel
from app.db.enums import AvailableCourse, AvailableDialect, AvailableLanguage


class CreateUserCourseProgressRequest(BaseModel):
    id: int
    course: AvailableCourse
    language: AvailableLanguage
    default_dialect: AvailableDialect | None
    total_modules: int

File: client/types/request_models/CreateUserCourseProgressRequest.ts
import { AvailableCourse, AvailableDialect, AvailableLanguage } from "../enums";

export default interface CreateUserCourseProgressRequest {
  id: number;
  course: AvailableCourse;
  language: AvailableLanguage;
  default_dialect: AvailableDialect | null;
  total_modules: number;
}
